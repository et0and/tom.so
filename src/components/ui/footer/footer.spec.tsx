import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import Footer from "./footer";

describe("<Footer/>", () => {
  it("renders correctly", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  it("displays the current year", () => {
    const { getByText } = render(<Footer />);
    const currentYear = new Date().getFullYear();
    expect(getByText(new RegExp(currentYear.toString()))).toBeInTheDocument();
  });

  it("contains accessibility and webring links", () => {
    const { getByText } = render(<Footer />);
    expect(getByText("Accessibility")).toHaveAttribute(
      "href",
      "/accessibility",
    );
    expect(getByText("webring")).toHaveAttribute(
      "href",
      "https://webring.xxiivv.com/#random",
    );
  });
});
