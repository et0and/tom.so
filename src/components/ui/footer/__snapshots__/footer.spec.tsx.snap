// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Footer/> > renders correctly 1`] = `
<div>
  <footer
    class="flex items-center tracking-tighter justify-between px-6 dark:bg-stone-900 dark:text-white py-4 lg:text-md sm:text-lg text-sm text-black"
  >
    <p>
      Â© 
      2025
       
      <a
        class="link"
        href="/accessibility"
      >
        Accessibility
      </a>
      . This site is part of a
       
      <a
        class="link"
        href="https://webring.xxiivv.com/#random"
      >
        webring
      </a>
      .
    </p>
  </footer>
</div>
`;
